src/main/java/com/zendesk/maxwell/schema/SchemaPosition.java

```diff
@@ -4,36 +4,104 @@
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.zendesk.maxwell.BinlogPosition;
 
-public class SchemaPosition {
-	public static BinlogPosition get(Connection c, Long serverID) throws SQLException {
-		PreparedStatement s = c.prepareStatement("SELECT * from `maxwell`.`positions` where server_id = ?");
-		s.setLong(1, serverID);
+public class SchemaPosition implements Runnable {
+	static final Logger LOGGER = LoggerFactory.getLogger(SchemaPosition.class);
+	private final Connection connection;
+	private final Long serverID;
+	private BinlogPosition lastPosition;
+	private final AtomicReference<BinlogPosition> position;
+	private final AtomicBoolean run;
+	private Thread thread;
 
-		ResultSet rs = s.executeQuery();
-		if ( !rs.next() )
-			return null;
+	public SchemaPosition(Connection c, Long serverID) {
+		this.connection = c;
+		this.serverID = serverID;
+		this.lastPosition = null;
+		this.position = new AtomicReference<>();
```

Comment:
this is java 7 syntax-fu, it implies the type of the generic from the variable type. 

to your second question: no.  Only the `.set()` and `.get()` calls are protected, so in *theory* someone could be mutating the `BinlogPosition` object underneath us.  In practice I think we can just not do that.